### PR TITLE
TST: fix compatibility pytest 8.0 (episode 2)

### LINKED ISSUE #15960: TST: CI is currently unstable against pytest 8.0
### Description

pytest 8.0.0 was just released a couple hours ago and despite our efforts to test against the release candidates, it still appears to break CI, but only certain test envs are affected.
I ran an experiment on my for at https://github.com/neutrinoceros/astropy/pull/13, with a no-op patch, so it's clear that failures are unrelated to the change.

I'm not 100% sure that pytest 8.0 is the culprit yet, but it's my primary (and currently, only) suspect.

### Expected behavior

_No response_

### How to Reproduce

_No response_

### Versions

_No response_

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description

A follow up to #15809
Fixes #15960 (🤞🏻)

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
Not all parts of #15960 are resolved yet, but I won't be able to contribute anything more until this Wednesday, so I'm undrafting this to let others merge it as is, if needed. Feel free to push to this branch while I'm away !

File: astropy/coordinates/tests/test_spectral_coordinate.py
Comment (eerovaher) [inline astropy/coordinates/tests/test_spectral_coordinate.py:1093]:
```suggestion
                    "No velocity defined on frame, assuming (0., 0., 0.) km / s.",
```

File: astropy/coordinates/tests/test_spectral_coordinate.py
Comment (eerovaher) [inline astropy/coordinates/tests/test_spectral_coordinate.py:1098]: I don't think I have ever seen an assert statement where the manually specified error message is better than the one `pytest` creates. Could you post a comparison?

Comment (neutrinoceros) [reply astropy/coordinates/tests/test_spectral_coordinate.py:1098]: Actually I can't, because I'm away and in reply-only mode until Wednesday. I wrote this in a hurry and didn't particularly think about how pytest would phrase the error if I let it. So feel free to patch this if you feel like it !

**Comment by pllim** [comment]
I think the wheel failure is a bug in version check.

```
          if PYTEST_LT_8_0:
              ctx = nullcontext()
          else:
              ctx = pytest.raises(VerifyWarning)
      
          with fits.open(path) as hdulist:
              with ctx, pytest.warns(wcs.FITSFixedWarning):
  >               w = wcs.WCS(hdulist[0].header)
  E               Failed: DID NOT RAISE <class 'astropy.io.fits.verify.VerifyWarning'>
  
  astropy/wcs/tests/test_wcs.py:1750: Failed
```

File: astropy/coordinates/tests/test_spectral_coordinate.py
Comment (pllim) [reply astropy/coordinates/tests/test_spectral_coordinate.py:1098]: I guess in pytest 8, now we have to catch 2 warnings instead of one? Let's get the jobs green first and then we can think about how to make it pretty.

Comment (pllim) [reply astropy/coordinates/tests/test_spectral_coordinate.py:1098]: Yay, CI is green! @eerovaher , would you prefer the style of handling this like `astropy/wcs/tests/test_wcs.py` (near the diff you touched) or you thinking about something else?

**Comment by pllim** [comment]
Let's also run the cron jobs, just in case. 🤞

File: astropy/coordinates/tests/test_spectral_coordinate.py
Comment (eerovaher) [reply astropy/coordinates/tests/test_spectral_coordinate.py:1098]: I don't see why we shouldn't use `pytest.warns()` to check for warnings here. Should I push a new commit?

**Comment by pllim** [comment]
Link check failure is unrelated (still waiting to hear back from OSU).

File: astropy/coordinates/tests/test_spectral_coordinate.py
Comment (pllim) [reply astropy/coordinates/tests/test_spectral_coordinate.py:1098]: Sure. Exotic archs are a bit slow but so far things are green, so I think this PR is ready for any clean-up and possibly merged soon. Thanks!

**Comment by eerovaher** [comment]
I have cleaned up the commit history to my liking. If all the commits would have been authored by a single person I would have combined all the patches into one, but that's not the case, so there are multiple commits to preserve authorship information.

As far as I can tell this should be good to merge, but because I am the committer of all the commits here then I shouldn't approve this formally.

File: astropy/coordinates/tests/test_spectral_coordinate.py
Comment (pllim) [reply astropy/coordinates/tests/test_spectral_coordinate.py:1083]: (Since oldestdeps has a pytest<8 pin,  I guess we will see in that job soon.)

Comment (eerovaher) [reply astropy/coordinates/tests/test_spectral_coordinate.py:1083]: I successfully tested this locally, but I had upgraded to `pytest` 8.0. When tried this with `pytest` 7.4 I indeed saw the failure you described.

Comment (pllim) [reply astropy/coordinates/tests/test_spectral_coordinate.py:1083]: Hmm weird, oldestdeps passed. Maybe this job isn't run there?

**Review by pllim — APPROVED** [review]
Diff LGTM now. Thanks, all!

I will not use auto-merge because I want to make sure the non-required jobs are also green before I merge.

File: astropy/coordinates/tests/test_spectral_coordinate.py
Comment (eerovaher) [reply astropy/coordinates/tests/test_spectral_coordinate.py]: This test has the `slow` marker, so it's skipped unless `pytest` is run with the `--run-slow` option.

**Pull request merged**
